app/uk/gov/hmrc/gform/gform/SectionRenderingService.scala

```diff
@@ -476,8 +510,54 @@ class SectionRenderingService(
         htmlForFileUpload(fieldValue, formTemplateId, index, ei, data, userDetails, maybeValidated, lang)
       case InformationMessage(infoType, infoText) =>
         htmlForInformationMessage(fieldValue, infoType, infoText, index, ei)
+      case HmrcTaxPeriod(idType, idNumber, regimeType) =>
+        htmlForHmrcTaxPeriod(
+          fieldValue,
+          index,
+          ei,
+          maybeValidated,
+          data,
+          obligations,
+          HmrcTaxPeriod(idType, idNumber, regimeType))
     }
 
+  private def htmlForHmrcTaxPeriod(
+    fieldValue: FormComponent,
+    index: Int,
+    ei: ExtraInfo,
+    validatedType: ValidatedType,
+    data: FormDataRecalculated,
+    obligations: Map[HmrcTaxPeriod, TaxPeriods],
+    hmrcTP: HmrcTaxPeriod) = {
+
+    val taxPeriodList = obligations
+      .get(hmrcTP) match {
+      case Some(c) => c.taxPeriods
+      case _       => List[TaxPeriod]()
+    }
+
+    val taxPeriodOptions = taxPeriodList
+      .map(
+        i =>
+          new OptionParams(
+            i.periodKey,
+            formatDate(i.inboundCorrespondenceFromDate),
+            formatDate(i.inboundCorrespondenceToDate),
+            false))
+    val validatedValue = buildFormFieldValidationResult(fieldValue, ei, validatedType, data)
+    val mapOfResultsOption = validatedValue match {
+      case Some(ComponentField(a, b)) => b
+      case _                          => Map[String, FormFieldValidationResult]()
+    }
+    val setValue = mapOfResultsOption.values.toList match {
+      case a => TextFormatter.formatText(Some(a(0))).split("|")(2)
```

Comment:
Access to `a(0)` is unsafe. Why don't use `case a :: _ => TextFormatter.formatText(Some(a)).split("|")(2)` instead ?